### ISSUE TITLE
Can't accept new commands. Controller is not running.

### ISSUE THREAD

**Issue by hychiang-git** [utecrobotics/ur5#5]
I get `Can't accept new commands. Controller is not running` error when I run `rosrun ur5_gazebo send_joints.py` in another terminal.

```
hychiang@hychiang-lab:~/ur5_grip/src/ur5$ roslaunch ur5_gazebo ur5_cubes.launch
... logging to /home/hychiang/.ros/log/a102a5e2-327a-11e9-bbc1-ac220b8958ba/roslaunch-hychiang-lab-12148.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://hychiang-lab:40237/

SUMMARY
========

PARAMETERS
 * /gazebo_ros_control/pid_gains/elbow_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/elbow_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/elbow_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/robotiq_85_left_knuckle_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/robotiq_85_left_knuckle_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/robotiq_85_left_knuckle_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/shoulder_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/shoulder_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/shoulder_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/shoulder_lift_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/shoulder_lift_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/shoulder_lift_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/shoulder_pan_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/shoulder_pan_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/shoulder_pan_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/wrist_1_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/wrist_1_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/wrist_1_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/wrist_2_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/wrist_2_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/wrist_2_joint/p: 1.0
 * /gazebo_ros_control/pid_gains/wrist_3_joint/d: 1.0
 * /gazebo_ros_control/pid_gains/wrist_3_joint/i: 0.0
 * /gazebo_ros_control/pid_gains/wrist_3_joint/p: 1.0
 * /gripper_controller/action_monitor_rate: 20
 * /gripper_controller/goal_tolerance: 0.002
 * /gripper_controller/joint: robotiq_85_left_k...
 * /gripper_controller/max_effort: 100
 * /gripper_controller/stall_timeout: 1.0
 * /gripper_controller/stall_velocity_threshold: 0.001
 * /gripper_controller/type: position_controll...
 * /joint_state_controller/publish_rate: 125
 * /joint_state_controller/type: joint_state_contr...
 * /robot_description: <?xml version="1....
 * /rosdistro: kinetic
 * /rosversion: 1.12.14
 * /state_publisher/publish_frequency: 50.0
 * /trajectory_controller/action_monitor_rate: 10
 * /trajectory_controller/constraints/elbow_joint/goal: 0.1
 * /trajectory_controller/constraints/elbow_joint/trajectory: 0.1
 * /trajectory_controller/constraints/goal_time: 0.6
 * /trajectory_controller/constraints/shoulder_lift_joint/goal: 0.1
 * /trajectory_controller/constraints/shoulder_lift_joint/trajectory: 0.1
 * /trajectory_controller/constraints/shoulder_pan_joint/goal: 0.1
 * /trajectory_controller/constraints/shoulder_pan_joint/trajectory: 0.1
 * /trajectory_controller/constraints/stopped_velocity_tolerance: 0.05
 * /trajectory_controller/constraints/wrist_1_joint/goal: 0.1
 * /trajectory_controller/constraints/wrist_1_joint/trajectory: 0.1
 * /trajectory_controller/constraints/wrist_2_joint/goal: 0.1
 * /trajectory_controller/constraints/wrist_2_joint/trajectory: 0.1
 * /trajectory_controller/constraints/wrist_3_joint/goal: 0.1
 * /trajectory_controller/constraints/wrist_3_joint/trajectory: 0.1
 * /trajectory_controller/joints: ['shoulder_pan_jo...
 * /trajectory_controller/state_publish_rate: 25
 * /trajectory_controller/stop_trajectory_duration: 0.5
 * /trajectory_controller/type: position_controll...
 * /use_sim_time: True

NODES
  /
    gazebo (gazebo_ros/gzserver)
    gazebo_gui (gazebo_ros/gzclient)
    gazebo_robot_controllers (controller_manager/spawner)
    spawn_gazebo_model (gazebo_ros/spawn_model)
    state_publisher (robot_state_publisher/state_publisher)

ROS_MASTER_URI=http://localhost:11311

process[gazebo-1]: started with pid [12168]
process[gazebo_gui-2]: started with pid [12173]
process[spawn_gazebo_model-3]: started with pid [12178]
process[gazebo_robot_controllers-4]: started with pid [12179]
process[state_publisher-5]: started with pid [12180]
[ INFO] [1550394466.020048309]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1550394466.020687930]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
[ INFO] [1550394466.103697777]: Finished loading Gazebo ROS API Plugin.
[ INFO] [1550394466.104388526]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...
SpawnModel script started
[INFO] [1550394466.404856, 0.000000]: Loading model XML from ros parameter
[INFO] [1550394466.411048, 0.000000]: Waiting for service /gazebo/spawn_urdf_model
[INFO] [1550394467.920364, 0.000000]: Calling service /gazebo/spawn_urdf_model
[INFO] [1550394468.249349, 0.000000]: Spawn status: SpawnModel: Successfully spawned entity
[INFO] [1550394468.249768, 0.000000]: Waiting for service /gazebo/set_model_configuration
[INFO] [1550394468.251373, 0.000000]: temporary hack to **fix** the -J joint position option (issue #93), sleeping for 1 second to avoid race condition.
[ INFO] [1550394468.377896611]: Loading gazebo_ros_control plugin
[ INFO] [1550394468.378024450]: Starting gazebo_ros_control plugin in namespace: /
[ INFO] [1550394468.379109520]: gazebo_ros_control plugin is waiting for model URDF in parameter [robot_description] on the ROS param server.
[ INFO] [1550394468.716116668]: Loaded gazebo_ros_control.
[ INFO] [1550394468.731512853]: ft_sensor plugin reporting wrench values to the frame [wrist_3_link]
[INFO] [1550394469.252763, 0.000000]: Calling service /gazebo/set_model_configuration
[INFO] [1550394469.255851, 0.000000]: Set model configuration status: SetModelConfiguration: success
[spawn_gazebo_model-3] process has finished cleanly
log file: /home/hychiang/.ros/log/a102a5e2-327a-11e9-bbc1-ac220b8958ba/spawn_gazebo_model-3*.log
[ERROR] [1550394582.275577101]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.275643693]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.275764450]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.275901255]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.275988830]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276058521]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276350526]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276448159]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276623309]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276726898]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276837135]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.276921407]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.277052603]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.277143055]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.277240088]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.277370089]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.277408213]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.278852640]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.278889331]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.279180039]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.279255993]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.279810802]: Can't accept new commands. Controller is not running.
[ERROR] [1550394582.280061004]: Can't accept new commands. Controller is not running.

```

I tried to check if my controller_manager is running, but still it appeared normal.
```
$ rosservice list | grep controller_manager
/controller_manager/list_controller_types
/controller_manager/list_controllers
/controller_manager/load_controller
/controller_manager/reload_controller_libraries
/controller_manager/switch_controller
/controller_manager/unload_controller
```
Could anyone helps?

Here are some related files:
- ur5gripper_controllers.yaml
- ur5_controllers.launch
- common.gazebo.xacro
- ur5_joint_limited_robot.urdf
- log file

**ur5gripper_controllers.yaml**
```
# Joint state controller (publish joint states)
joint_state_controller:
  publish_rate: 125
  type: joint_state_controller/JointStateController

# Gripper controller
gripper_controller:
  type: position_controllers/GripperActionController
  joint: robotiq_85_left_knuckle_joint
  action_monitor_rate: 20
  goal_tolerance: 0.002
  max_effort: 100
  stall_velocity_threshold: 0.001
  stall_timeout: 1.0

# Trajectory controller
trajectory_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - shoulder_pan_joint
    - shoulder_lift_joint
    - elbow_joint
    - wrist_1_joint
    - wrist_2_joint
    - wrist_3_joint
  constraints:
    goal_time: 0.6
    stopped_velocity_tolerance: 0.05
    shoulder_pan_joint: {trajectory: 0.1, goal: 0.1}
    shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
    elbow_joint: {trajectory: 0.1, goal: 0.1}
    wrist_1_joint: {trajectory: 0.1, goal: 0.1}
    wrist_2_joint: {trajectory: 0.1, goal: 0.1}
    wrist_3_joint: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate: 25
  action_monitor_rate: 10


# gazebo_ros_control_params.yaml
gazebo_ros_control/pid_gains:
  shoulder_joint: {p: 1.0, i: 0.0, d: 1.0}
  shoulder_lift_joint: {p: 1.0, i: 0.0, d: 1.0}
  shoulder_pan_joint: {p: 1.0, i: 0.0, d: 1.0}
  elbow_joint: {p: 1.0, i: 0.0, d: 1.0}
  wrist_1_joint: {p: 1.0, i: 0.0, d: 1.0}
  wrist_2_joint: {p: 1.0, i: 0.0, d: 1.0}
  wrist_3_joint: {p: 1.0, i: 0.0, d: 1.0}
  robotiq_85_left_knuckle_joint: {p: 1.0, i: 0.0, d: 1.0}
```

**ur5_controllers.launch**
```
<?xml version="1.0"?>
<launch>
  <!-- Launch file parameters -->
  <arg name="debug"     default="false" />

  <arg if=      "$(arg debug)"  name="DEBUG" value="screen"/>
  <arg unless = "$(arg debug)"  name="DEBUG" value="log"/>

  <!-- Controllers config -->
  <rosparam file="$(find ur5_gazebo)/controller/ur5gripper_controllers.yaml"
            command="load" />

  <!-- Load controllers -->
  <node name="gazebo_robot_controllers" pkg="controller_manager" type="spawner"
        output="$(arg DEBUG)"
        args="joint_state_controller trajectory_controller gripper_controller"/>

  <!-- TF -->
  <node pkg="robot_state_publisher" type="state_publisher"
        name="state_publisher">
    <param name="publish_frequency"  type="double" value="50" />
  </node>

</launch>
```

**common.gazebo.xacro**
```
<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
      <legacyModeNS>true</legacyModeNS>
    </plugin>

<!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
  </gazebo>

</robot>

```

**ur5_joint_limited_robot.urdf**
```
<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ur5_joint_limited_robot.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur5" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
    <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
  </gazebo>
  <!-- measured from model -->
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.089159"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.13585 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28"/>
      <inertia ixx="0.22689067591" ixy="0.0" ixz="0.0" iyy="0.22689067591" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1197 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
      <inertia ixx="0.049443313556" ixy="0.0" ixz="0.0" iyy="0.049443313556" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.39225"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.093 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.09465"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur5_description/meshes/ur5/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0171364731454" ixy="0.0" ixz="0.0" iyy="0.0171364731454" iyz="0.0" izz="0.033822"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.0823 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- nothing to do here at the moment -->
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0"/>
  <joint name="wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.0823 0"/>
    <parent link="wrist_3_link"/>
    <child link="tool0"/>
  </joint>
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
</robot>
```

** Log file **
```
[rospy.client][INFO] 2019-02-17 17:30:57,490: init_node, name[/spawn_gazebo_model], pid[14504]
[xmlrpc][INFO] 2019-02-17 17:30:57,490: XML-RPC server binding to 0.0.0.0:0
[xmlrpc][INFO] 2019-02-17 17:30:57,490: Started XML-RPC server [http://hychiang-lab:35709/]
[rospy.init][INFO] 2019-02-17 17:30:57,490: ROS Slave URI: [http://hychiang-lab:35709/]
[rospy.impl.masterslave][INFO] 2019-02-17 17:30:57,491: _ready: http://hychiang-lab:35709/
[rospy.registration][INFO] 2019-02-17 17:30:57,491: Registering with master node http://localhost:11311
[xmlrpc][INFO] 2019-02-17 17:30:57,491: xml rpc node: starting XML-RPC server
[rospy.init][INFO] 2019-02-17 17:30:57,591: registered with master
[rospy.rosout][INFO] 2019-02-17 17:30:57,591: initializing /rosout core topic
[rospy.rosout][INFO] 2019-02-17 17:30:57,593: connected to core topic /rosout
[rospy.simtime][INFO] 2019-02-17 17:30:57,595: initializing /clock core topic
[rospy.simtime][INFO] 2019-02-17 17:30:57,597: connected to core topic /clock
[rosout][INFO] 2019-02-17 17:30:57,600: Loading model XML from ros parameter
[rosout][INFO] 2019-02-17 17:30:57,605: Waiting for service /gazebo/spawn_urdf_model
[rospy.internal][INFO] 2019-02-17 17:30:57,835: topic[/rosout] adding connection to [/rosout], count 0
[rospy.internal][INFO] 2019-02-17 17:30:59,071: topic[/clock] adding connection to [http://hychiang-lab:35641/], count 0
[rosout][INFO] 2019-02-17 17:30:59,114: Calling service /gazebo/spawn_urdf_model
[rosout][INFO] 2019-02-17 17:30:59,325: Spawn status: SpawnModel: Successfully spawned entity
[rosout][INFO] 2019-02-17 17:30:59,326: Waiting for service /gazebo/set_model_configuration
[rosout][INFO] 2019-02-17 17:30:59,328: temporary hack to **fix** the -J joint position option (issue #93), sleeping for 1 second to avoid race condition.
[rosout][INFO] 2019-02-17 17:31:00,330: Calling service /gazebo/set_model_configuration
[rosout][INFO] 2019-02-17 17:31:00,333: Set model configuration status: SetModelConfiguration: success
[rospy.core][INFO] 2019-02-17 17:31:00,333: signal_shutdown [atexit]
[rospy.internal][INFO] 2019-02-17 17:31:00,335: topic[/rosout] removing connection to /rosout
[rospy.internal][INFO] 2019-02-17 17:31:00,335: topic[/clock] removing connection to http://hychiang-lab:35641/
[rospy.internal][WARNING] 2019-02-17 17:31:00,336: Unknown error initiating TCP/IP socket to hychiang-lab:45963 (http://hychiang-lab:35641/): Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_base.py", line 557, in connect
    self.read_header()
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_base.py", line 650, in read_header
    self._validate_header(read_ros_handshake_header(sock, self.read_buff, self.protocol.buff_size))
AttributeError: 'NoneType' object has no attribute 'buff_size'

[rospy.topics][ERROR] 2019-02-17 17:31:00,336: Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 326, in close
    c.close()
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_base.py", line 838, in close
    self.socket.close()
AttributeError: 'NoneType' object has no attribute 'close'

[rospy.impl.masterslave][INFO] 2019-02-17 17:31:00,336: atexit

```

**Comment by hychiang-git** [comment]
I followed and went through the tutorial here: http://gazebosim.org/tutorials/?tut=ros_control#Tutorial:ROSControl
Therefore, I also tried to create a control directory under my workspace.

I also tried to launch a controller in the other terminal, but still got the same error.
At the first terminal, I ran: `roslaunch ur5_gazebo ur5_cubes.launch`
At the second terminal, I ran: `roslaunch ur5_gazebo ur5_controllers.launch`
Finally, at the third termainl, I ran `rosrun ur5_gazebo send_gripper.py --value 0.5`

However, I still got the same error `[ERROR] [1550456337.108503510]: Can't accept new action goals. Controller is not running.`
